### ISSUE TITLE
Hub 2245 not connecting?

### ISSUE THREAD

**Issue by egrass-git** [kuestess/homebridge-platform-insteonlocal#125]
Hi, I am newly using Homebridge and Insteonlocal on a Mac (Catalina) to try to operate a Hub 2245.  No other plugins are installed.

Everything appears to be set up ok.  Indeed, there are controllable devices in Home.  However, it seems the the Hub is not receiving or issuing commands when I trigger those Home icons or when I trigger the device icons in Homebridge.

I did double and triple check that the user name and password are for the Hub not the Insteon software.  (I changed the Hub default).  If I use the Insteon software, the Hub triggers the devices just fine.

One hint might be that this does not work so far as I can tell, it just times out so I put all the devices in manually:
"Direct your browser to the host that you have Insteonlocal running on and the port configured in your config.json (ie, 127.0.0.1:3000 if running on the local machine)."  I did try changing this to my Mac IP address etc and still nothing.

Below is my config file and a log.  Any help greatly appreciated!!!  This looks so great, really hope to get it working for me.

{
    "bridge": {
        "name": "Homebridge ED38",
        "username": "0E:D6:B4:DC:ED:38",
        "port": 51755,
        "pin": "128-26-815"
    },
    "accessories": [],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "auth": "form",
            "theme": "auto",
            "tempUnits": "f",
            "lang": "auto",
            "platform": "config"
        },
        {
            "platform": "InsteonLocal",
            "name": "Insteon Local Platform",
            "user": "xyz",
            "pass": "abcd",
            "host": "192.168.1.29.83",
            "port": "25105",
            "model": "2245",
            "refresh": "0",
            "use_express": true,
            "server_port": "3000",
            "keepAlive": "3600",
            "devices": [
                {
                    "name": "Library Lights",
                    "deviceID": "4A.0D.AF",
                    "dimmable": "yes",
                    "deviceType": "dimmer"
                },
                {
                    "name": "Library Fan",
                    "deviceID": "51.1A.DB",
                    "deviceType": "switch"
                },
                {
                    "name": "Front Light",
                    "deviceID": "44.3A.44",
                    "dimmable": "yes",
                    "deviceType": "switch"
                }
            ]
        }
    ]
}

[12/11/2020, 12:44:53 AM] [Homebridge UI] Changes to config.json saved.
[12/11/2020, 12:44:56 AM] [Homebridge UI] Homebridge restart request received
[12/11/2020, 12:44:56 AM] [Homebridge UI] UI / Bridge settings have not changed; only restarting Homebridge process
[12/11/2020, 12:44:56 AM] [HB Supervisor] Sending SIGTERM to Homebridge
[12/11/2020, 12:44:56 AM] Got SIGTERM, shutting down Homebridge...
[12/11/2020, 12:45:01 AM] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[12/11/2020, 12:45:07 AM] [HB Supervisor] Restarting Homebridge...
[12/11/2020, 12:45:07 AM] [HB Supervisor] Starting Homebridge with extra flags: -I
[12/11/2020, 12:45:07 AM] [HB Supervisor] Started Homebridge v1.1.6 with PID: 18777
[12/11/2020, 12:45:07 AM] Loaded config.json with 0 accessories and 2 platforms.
[12/11/2020, 12:45:07 AM] ---
[12/11/2020, 12:45:07 AM] Loaded plugin: homebridge-config-ui-x@4.36.0
[12/11/2020, 12:45:07 AM] Registering platform 'homebridge-config-ui-x.config'
[12/11/2020, 12:45:07 AM] ---
[12/11/2020, 12:45:07 AM] Loaded plugin: homebridge-platform-insteonlocal@0.4.15
homebridge API version: 2.6
[12/11/2020, 12:45:07 AM] Registering platform 'homebridge-platform-insteonlocal.InsteonLocal'
[12/11/2020, 12:45:07 AM] ---
[12/11/2020, 12:45:07 AM] Loading 2 platforms...
[12/11/2020, 12:45:07 AM] [Config] Initializing config platform...
[12/11/2020, 12:45:07 AM] [Config] Running in Service Mode
[12/11/2020, 12:45:07 AM] [Insteon Local Platform] Initializing InsteonLocal platform...
Initializing Insteon UI 
Reading config from /Users/egrass/.homebridge/config.json
Found 2 platform(s) in config
Creating new insteon.json
No devices in config
[12/11/2020, 12:45:07 AM] [Insteon Local Platform] Started connection watcher...
[12/11/2020, 12:45:07 AM] [Insteon Local Platform] Found 3 devices in config
[12/11/2020, 12:45:07 AM] Initializing platform accessory 'Library Lights'...
[12/11/2020, 12:45:07 AM] Initializing platform accessory 'Library Fan'...
[12/11/2020, 12:45:07 AM] Initializing platform accessory 'Front Light'...
Setup Payload:
X-HM://0023OKHBZ9FD4
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 123-45-678 │     
    └────────────┘     
                       
[12/11/2020, 12:45:07 AM] Homebridge is running on port 51755.
[12/11/2020, 12:46:06 AM] [Insteon Local Platform] Setting power state of Library Fan to on
[12/11/2020, 12:46:06 AM] [Insteon Local Platform] Setting level of Library Lights to 100%
[12/11/2020, 12:46:06 AM] [Insteon Local Platform] Setting power state of Front Light to on
HAP Warning: Characteristic 00000008-0000-1000-8000-0026BB765291 not in required or optional characteristics for service 00000049-0000-1000-8000-0026BB765291. Adding anyway.
[12/11/2020, 12:46:07 AM] [Insteon Local Platform] Setting power state of Front Light to off
[12/11/2020, 12:46:07 AM] [Insteon Local Platform] Setting power state of Library Fan to off
[12/11/2020, 12:46:26 AM] [Insteon Local Platform] Error setting power state of Library Fan to on
[12/11/2020, 12:46:26 AM] [Insteon Local Platform] Getting status for Library Fan
[12/11/2020, 12:46:36 AM] [Insteon Local Platform] Error setting power state of Front Light to off
[12/11/2020, 12:46:36 AM] [Insteon Local Platform] Getting status for Front Light
[12/11/2020, 12:46:47 AM] [Insteon Local Platform] Error setting power state of Front Light to on
[12/11/2020, 12:46:47 AM] [Insteon Local Platform] Getting status for Front Light
[12/11/2020, 12:46:57 AM] [Insteon Local Platform] Error setting power state of Library Fan to off
[12/11/2020, 12:46:57 AM] [Insteon Local Platform] Getting status for Library Fan
[12/11/2020, 12:47:07 AM] [Insteon Local Platform] Error getting power state of Library Fan
[12/11/2020, 12:47:17 AM] [Insteon Local Platform] Error getting power state of Front Light
[12/11/2020, 12:47:28 AM] [Insteon Local Platform] Error getting power state of Front Light
[12/11/2020, 12:47:38 AM] [Insteon Local Platform] Error getting power state of Library Fan
[12/11/2020, 1:10:15 AM] [Homebridge UI] Running job to cleanup config.json backup files older than 60 days...
[12/11/2020, 1:15:08 AM] [Homebridge UI] Running scheduled instance backup...
[12/11/2020, 1:15:08 AM] [Homebridge UI] Creating temporary backup archive at /var/folders/f_/z1jjpfq924j2dt6fv8r1xhsw0000gn/T/homebridge-backup-9mInk3/homebridge-backup-0ED6B4DCED38.tar.gz
[12/11/2020, 1:45:07 AM] [Insteon Local Platform] Closing connection to Hub...
[12/11/2020, 1:45:12 AM] [Insteon Local Platform] Reconnecting to Hub...
[12/11/2020, 2:45:07 AM] [Insteon Local Platform] Closing connection to Hub...
[12/11/2020, 2:45:12 AM] [Insteon Local Platform] Reconnecting to Hub...
[12/11/2020, 3:45:07 AM] [Insteon Local Platform] Closing connection to Hub...
[12/11/2020, 3:45:12 AM] [Insteon Local Platform] Reconnecting to Hub...
[12/11/2020, 4:45:07 AM] [Insteon Local Platform] Closing connection to Hub...
[12/11/2020, 4:45:12 AM] [Insteon Local Platform] Reconnecting to Hub...
[12/11/2020, 5:45:07 AM] [Insteon Local Platform] Closing connection to Hub...
[12/11/2020, 5:45:12 AM] [Insteon Local Platform] Reconnecting to Hub...
[12/11/2020, 6:45:07 AM] [Insteon Local Platform] Closing connection to Hub...
[12/11/2020, 6:45:12 AM] [Insteon Local Platform] Reconnecting to Hub...
[12/11/2020, 7:45:07 AM] [Insteon Local Platform] Closing connection to Hub...
[12/11/2020, 7:45:12 AM] [Insteon Local Platform] Reconnecting to Hub...
[12/11/2020, 8:45:07 AM] [Insteon Local Platform] Closing connection to Hub...
[12/11/2020, 8:45:12 AM] [Insteon Local Platform] Reconnecting to Hub...
[12/11/2020, 9:45:07 AM] [Insteon Local Platform] Closing connection to Hub...
[12/11/2020, 9:45:12 AM] [Insteon Local Platform] Reconnecting to Hub...

**Comment by kuestess** [comment]
@egrass-git Looks like the plugin isn't connecting to the hub.  You should see a 'Connected to Insteon 2245 Hub' in the log.  Verify the IP/port and then double check your username/password.  The user/pass is not the email/password you use to initially log into the Insteon app - it the actual Hub user/pass.  In the Insteon app, go to Settings -> House and use the username and password at the bottom of the screen.

**Comment by egrass-git** [comment]
That worked, but what is weird is I had seen the posts about people screwing this up and had gotten the correct user name and password for the Hub before NOT the Insteon app.  Did not work as per my last post.

After reading your post, I went to the Hub via Safari to 198.168.1.83:2150 and after a long time it game me user name and password to fill in.  Did not work!  So in the Insteon app I changed the password to "password" (no quotes).  And that worked for the direct connect.

I then changed the password in insteonlocal to password and it worked!  I got the proper Connected to Insteon 2245 Hub response.

I realize I could have misspelled the Hub user name or password before but I am 99% sure I did not (and tried several different user names and passwords).

Weird.
